src/Util/CsvImportUtil.php

```diff
@@ -0,0 +1,428 @@
+        }
+
+        $this->storeInDatabase($field, $this->getData());
+    }
+
+    /**
+     * Return true if the form was submitted
+     *
+     * @return bool
+     */
+    public function isFormSubmitted()
+    {
+        return $this->request->request->get('FORM_SUBMIT') === $this->getFormId();
+    }
+
+    /**
+     * Get the uploaded files
+     *
+     * @return array
+     */
+    public function getUploadedFiles()
+    {
+        $files = [];
+
+        foreach ($this->getUploader()->uploadTo($this->getUploadFolder()) as $filePath) {
+            $file = new File($filePath);
+
+            // Add an error if the file extension is not valid
+            if (!in_array($file->extension, $this->getFileExtensions(), true)) {
+                $this->request->getSession()->getFlashBag()->add(
+                    'error',
+                    sprintf($GLOBALS['TL_LANG']['ERR']['filetype'], $file->extension)
+                );
+
+                continue;
+            }
+
+            $files[] = $filePath;
+        }
+
+        return $files;
+    }
+
+    /**
+     * Get the data from uploaded files
+     *
+     * @throws CsvImportErrorException
+     */
+    public function getData()
+    {
+        $files = $this->getUploadedFiles();
+
+        // Add an error and reload the page if there was no file selected
+        if (count($files) === 0) {
+            throw new CsvImportErrorException($GLOBALS['TL_LANG']['ERR']['all_fields']);
+        }
+
+        return $this->getDataFromFiles($files);
+    }
+
+    /**
+     * Store data in the database
+     *
+     * @param string $field
+     * @param array  $data
+     *
+     * @throws \Doctrine\DBAL\DBALException
+     */
+    public function storeInDatabase($field, array $data)
+    {
+        $id    = $this->request->query->get('id');
+        $table = $this->request->query->get('table');
+
+        $versions = new Versions($table, $id);
+        $versions->create();
+
+        $this->connection->prepare("UPDATE $table SET $field=? WHERE id=?")
+            ->execute([serialize($data), $id]);
+    }
+
+    /**
+     * Get data callback
+     *
+     * @return callable
+     */
+    public function getCallback()
+    {
+        return $this->callback;
+    }
+
+    /**
+     * Set data callback
+     *
+     * @param callable $callback
+     */
+    public function setCallback(callable $callback)
+    {
+        $this->callback = $callback;
+    }
+
+    /**
+     * Get separators
+     *
+     * @return array
+     */
+    public function getSeparators()
+    {
+        return $this->separators;
+    }
+
+    /**
+     * Set separators
+     *
+     * @param array $separators
+     */
+    public function setSeparators(array $separators)
+    {
+        $this->separators = $separators;
+    }
+
+    /**
+     * Get the template
+     *
+     * @return Template
+     */
+    public function getTemplate()
+    {
+        if ($this->template === null) {
```

Comment:
Yeah, I don't feel like changing everything now for no reason. We'll keep that in mind for the future though.